python/calc_refractive_index.py

```diff
@@ -23,99 +23,193 @@
 from __future__ import print_function, division, absolute_import
 import numpy as np
 
```

Comment:
A general comment on the functions in this file: *if* you plan to call these functions holding most of the quantities constant while varying just one or two of them (e.g. wavelength) then I suggest you implement this as a class. The user constructs the object specifying the stationary values, which are saved as instance variables. The methods then need only take arguments for the few things the user varies.

However this only makes sense if you want to call some of these functions repeatedly while holding a well defined subset of parameters constant.